### PR TITLE
Add predict_proba output option to SklearnModelProcessor

### PR DESCRIPTION
**Pull request by santiemckenzie**
Add a predict_method setting ("predict" | "predict_proba", default "predict") so the processor can emit a per-class posterior vector instead of only a hard label. When set to "predict_proba", the output ch axis is relabeled to the model's classes_. Supports sklearn predict_proba and River predict_proba_many. The default predict path is unchanged.

Motivated by consumers that pool per-bin posteriors over a window (see BlackrockNeurotech/intent-pipelines#13), which previously required a bespoke decode unit because SklearnModelProcessor only exposed predict.

### CODE REVIEW

**Comment by cboulay** [comment]
Thanks for the PR — this is well put together. The test coverage is appreciated, especially the default-behavior regression test, and the motivation makes sense.

One requested change before merge: please convert `predict_method` from a bare `str` to a `(str, Enum)`, matching the existing convention in `src/ezmsg/learn/util.py` (`RegressorType`, `AdaptiveLinearRegressor`, etc.). This keeps the door open for future inference methods (`decision_function`, `predict_log_proba`, ...) without an API break, and because it subclasses `str`, plain-string configs keep working.

In `process/sklearn.py`:

```python
from enum import Enum


class PredictMethod(str, Enum):
    PREDICT = "predict"
    PREDICT_PROBA = "predict_proba"
```

Settings:

```python
predict_method: PredictMethod = PredictMethod.PREDICT
```

Comparisons then read as:

```python
if self.settings.predict_method == PredictMethod.PREDICT_PROBA:
```

which also works when the setting was supplied as a plain string like `"predict_proba"`. With that in place, the hand-rolled membership check in `_reset_state` can be reduced to a one-line coercion (`PredictMethod(self.settings.predict_method)`) or dropped.

One thing I noticed that's out of scope for this PR: with River classifiers the `ch` axis falls back to `np.arange` labels even though `predict_proba_many` returns the actual class labels as DataFrame columns. I've opened #17 to track that — no action needed here.

**Comment by santiemckenzie** [comment]
Made those changes: `predict_method` is now a `(str, Enum)`, and the check in `_reset_state` is down to the one-line coercion. Thank you!
